modules/tensor_mechanics/src/postprocessors/SidesetForce.C

```diff
@@ -0,0 +1,46 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SidesetForce.h"
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("TensorMechanicsApp", SidesetForce);
+registerMooseObject("TensorMechanicsApp", ADSidesetForce);
+
+template <bool is_ad>
+InputParameters
+SidesetForceTempl<is_ad>::validParams()
+{
+  InputParameters params = SideIntegralPostprocessor::validParams();
+  params.addClassDescription(
+      "This postprocessor computes the force on sidesets in parameter direction");
+  params.addRequiredParam<MaterialPropertyName>("stress_tensor", "The rank two stress tensor name");
+  params.addRequiredParam<RealVectorValue>("direction",
+                                           "Direction in which the force is to be computed");
+  params.set<bool>("use_displaced_mesh") = true;
+  return params;
+}
+
+template <bool is_ad>
+SidesetForceTempl<is_ad>::SidesetForceTempl(const InputParameters & parameters)
+  : SideIntegralPostprocessor(parameters),
+    _tensor(getGenericMaterialProperty<RankTwoTensor, is_ad>("stress_tensor")),
```

Comment:
FYI, we're pretty sure there's a bug in `getGenericMaterialProperty`. At least for volume integration points, it is pretty clearly giving us values for the wrong integration point. We just observed this, and haven't opened an issue yet. It may be working fine for you here, but if it's doing something weird, you could be hitting that too.